### PR TITLE
Added some validation to the nickname

### PR DESCRIPTION
**Pull request by ludoviccr**
#111 adding some validation to the nickname and front end errros

### CODE REVIEW

File: app/views/nicknames/new.html.erb [hunk 1]
```diff
@@ -1,3 +1,6 @@
+<%= @nickname.errors.full_messages.first if @nickname.errors.any? %>
```
Comment (gmanningcr) [inline app/views/nicknames/new.html.erb]: We're only going to see the first error with this and there may be more than one validation that prevents the nickname from being saved. We should display each error that is returned. Something like the below.

```
<% if @nickname.errors.any? %>
	<div id="error_reason">
	<h2>
		<%= pluralize(@nickname.errors.count, "error") %> prevented this nickname from being saved:
	</h2>
	<ul>
		<%= @nickname.errors.full_messages.each do |message| %>
			<li><%= message %></li>
		<% end %>
	</ul>
	</div>
<% end %>
```

File: app/models/nickname.rb [hunk 2]
```diff
@@ -1,4 +1,23 @@
...
+  	'omnipotent',
+  	'omniscient',
+  	'nazi',
+  	'hitler'
+  ]
+  @exclusions_for_regex = Regexp.new '(' + @exclusions.join("|").to_s + ')'
+  validates_format_of :nickname,{
+  	without: @exclusions_for_regex,
+  	message: "do not think you are %{value}"
+  }
```
Comment (gmanningcr) [inline app/models/nickname.rb]: This does seem a bit clunky, was there a reason not to use validates_exclusion_of? At the moment we're denying any string that contains any of these words, so we couldn't use "The Godfather" for example. If we do want to continue to exclude all strings that contain any of these strings, then we can continue with validates_format_of, but we could probably make this a bit easier to read if so.

File: db/schema.rb [hunk 3]
```diff
@@ -10,7 +10,11 @@
 #
 # It's strongly recommended that you check this file into your version control system.
 
-ActiveRecord::Schema.define(version: 20180321123300) do
+ActiveRecord::Schema.define(version: 20180413175359) do
+
+  create_table "banned_phrases", force: :cascade do |t|
+    t.string "phrase"
+  end
```
Comment (gmanningcr) [inline db/schema.rb]: This needs to be redone as a migration, we shouldn't be changing the schema.rb file directly.

Comment (cr-abednorz) [reply db/schema.rb]: I have no idea how I achieved that :( I thought I made a migration for this

Comment (cr-abednorz) [reply db/schema.rb]: I'll add a migration in to support this and push again

**Pull request merged**